docusaurus/docs/Android/02-client/06-guides/11-listening-for-events.mdx

```diff
@@ -0,0 +1,45 @@
+# Listening for Events
+
+As soon as you call `watchChannel` on a Channel or `queryChannels`, the SDK will start receiving events to stay up to date with changes. You can hook into specific events:
+
+```kotlin
+val channelClient = client.channel("messaging", "general")
+
+// Subscribe for new message events
+val disposable: Disposable = channelClient.subscribeFor<NewMessageEvent> { newMessageEvent ->
+    val message = newMessageEvent.message
+}
+
+// Dispose when you want to stop receiving events
+disposable.dispose()
+```
```

Comment:
Just for curiosity, when do we call the `dispose` method typically in the lifecycle? (onPause or onDestroy)

Reply:
That's actually a very good question. If the subscription was done in `onViewCreated`, then I would do dispose in `onDestroyView`.

Reply:
Thanks! I'm wondering if I can work on auto-dispose features based on the Android lifecycle for the subscription.